skimage/measure/profile.py

```diff
@@ -50,17 +53,27 @@ def profile_line(img, src, dst, linewidth=1,
     The destination point is included in the profile, in contrast to
     standard numpy indexing.
     """
-    perp_lines = _line_profile_coordinates(src, dst, linewidth=linewidth)
-    if img.ndim == 3:
-        pixels = [ndi.map_coordinates(img[..., i], perp_lines,
-                                      order=order, mode=mode, cval=cval)
-                  for i in range(img.shape[2])]
+    if img.ndim == 4 and multichannel:
+        # 3D RGB to be implemented
+        raise NotImplementedError('3D RGB to be implemented')
+    elif img.ndim == 3 and not multichannel:
+        # 3D intensity
+        perp_lines = _line_profile_coordinates3d(src, dst, linewidth=linewidth)
+        pixels = ndi.map_coordinates(img, perp_lines, order=order, mode=mode, cval=cval)
+    elif img.ndim == 3 and multichannel:
+        # 2D RGB
+        perp_lines = _line_profile_coordinates(src, dst, linewidth=linewidth)
+        pixels = [ndi.map_coordinates(img[..., i], perp_lines, order=order,
+                                      mode=mode, cval=cval) for i in range(img.shape[2])]
         pixels = np.transpose(np.asarray(pixels), (1, 2, 0))
+    elif img.ndim == 2:
+        # 2D intensity
+        perp_lines = _line_profile_coordinates(src, dst, linewidth=linewidth)
+        pixels = ndi.map_coordinates(img, perp_lines, order=order, mode=mode, cval=cval)
     else:
-        pixels = ndi.map_coordinates(img, perp_lines,
-                                     order=order, mode=mode, cval=cval)
-    intensities = pixels.mean(axis=1)
+        raise ValueError('Invalid arguments')
```

Comment:
Try to explain *why* they are invalid. e.g. `'profile_line is not implemented for images of dimension %s' % img.shape`